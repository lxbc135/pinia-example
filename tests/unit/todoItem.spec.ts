import { createPinia, setActivePinia } from 'pinia'
import { createApp } from 'vue';
import { mount } from '@vue/test-utils'
import TodoItem from '@/components/TodoItem.vue';

describe('TodoItem', () => {
  const app = createApp({});
  const pinia = createPinia();

  beforeEach(() => {
    app.use(pinia);
    setActivePinia(pinia);
  });

  it('should init todo item from props', () => {
    const wrapper = mount(TodoItem, {
      props: {
        todo: {id: 0, content: 'todo1', isCompleted: false}
      }
    });
    expect(wrapper.get('li').text()).toContain('todo1');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
