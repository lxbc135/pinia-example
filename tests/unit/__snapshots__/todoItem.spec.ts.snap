// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem should init todo item from props 1`] = `
"<li class=\\"flex items-center justify-between p-2 border-2 border-solid border-gray-300 rounded-lg my-1\\"><span class=\\"text-gray-900 text-lg break-all\\">todo1</span>
  <div class=\\"ml-2 flex gap-1\\"><button class=\\"p-2 border-solid border-2 border-gray-400 rounded-lg text-gray-500 hover:text-green-500 hover:border-green-500\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"h-6 w-6\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"2\\">
        <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" d=\\"M5 13l4 4L19 7\\"></path>
      </svg></button><button class=\\"p-2 border-solid border-2 border-gray-400 rounded-lg text-gray-500 hover:text-red-500 hover:border-red-500\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"h-6 w-6\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"2\\">
        <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" d=\\"M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16\\"></path>
      </svg></button></div>
</li>"
`;
